### DISCUSSION TITLE
Create interactive instance failed

### DISCUSSION THREAD

**Discussion by default233** [alibaba/GraphScope#1046] [Q&A] [answered]
执行示例论文异构分类任务时，出现以下问题
interactive = sess.gremlin(graph)

2021-11-25 11:05:37,271 [ERROR][rpc:223]: Runstep failed with code: INTERACTIVE_ENGINE_INTERNAL_ERROR, message: Error occurred during preprocessing, The traceback is: Traceback (most recent call last):
  File "\~/.local/lib/python3.8/site-packages/gscoordinator/coordinator.py", line 738, in _create_interactive_instance
    raise RuntimeError(
RuntimeError: Error code: 127, message /usr/bin/env: ‘bash\r’: No such file or directory

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "\~/.local/lib/python3.8/site-packages/gscoordinator/coordinator.py", line 511, in RunStep
    self.run_on_interactive_engine(
  File "\~/.local/lib/python3.8/site-packages/gscoordinator/coordinator.py", line 428, in run_on_interactive_engine
    op_result = self._create_interactive_instance(op)
  File "\~/.local/lib/python3.8/site-packages/gscoordinator/coordinator.py", line 744, in _create_interactive_instance
    raise RuntimeError("Create interactive instance failed.") from e
RuntimeError: Create interactive instance failed.

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
\~/.local/lib/python3.8/site-packages/graphscope/client/session.py in gremlin(self, graph, engine_params)
   1144             enable_gaia = self._config_params["enable_gaia"]
-> 1145             _wrapper = self._wrapper(
   1146                 InteractiveQueryDAGNode(self, graph, engine_params, enable_gaia)

\~/.local/lib/python3.8/site-packages/graphscope/client/session.py in _wrapper(self, dag_node)
    921         if self.eager():
--> 922             return self.run(dag_node)
    923         else:

\~/.local/lib/python3.8/site-packages/graphscope/client/session.py in run(self, fetches, debug)
    950         try:
--> 951             response = self._grpc_client.run(fetch_handler.targets)
    952         except FatalError:

\~/.local/lib/python3.8/site-packages/graphscope/client/rpc.py in run(self, dag_def)
    145     def run(self, dag_def):
--> 146         return self._run_step_impl(dag_def)
    147 

\~/.local/lib/python3.8/site-packages/graphscope/client/rpc.py in with_grpc_catch(*args, **kwargs)
     46         try:
---> 47             return fn(*args, **kwargs)
     48         except grpc.RpcError as exc:

\~/.local/lib/python3.8/site-packages/graphscope/client/rpc.py in _run_step_impl(self, dag_def)
    228             if response.full_exception:
--> 229                 raise pickle.loads(response.full_exception)
    230         return response

RuntimeError: Create interactive instance failed.

The above exception was the direct cause of the following exception:

InteractiveEngineInternalError            Traceback (most recent call last)
/tmp/ipykernel_16037/1581575573.py in <module>
      1 # get the entrypoint for submitting Gremlin queries on graph g.
----> 2 interactive = sess.gremlin(graph)
      3 
      4 # count the number of papers two authors (with id 2 and 4307) have co-authored.
      5 papers = interactive.execute("g.V().has('author', 'id', 2).out('writes').where(__.in('writes').has('id', 4307)).count()").one()

~/.local/lib/python3.8/site-packages/graphscope/client/utils.py in wrapper(*args, **kwargs)
    184             func.__defaults__ = tuple(new_defaults)
    185 
--> 186             return_value = func(*args, **kwargs)
    187 
    188             # Restore original defaults.

~/.local/lib/python3.8/site-packages/graphscope/client/session.py in gremlin(self, graph, engine_params)
   1150                 interactive_query.status = InteractiveQueryStatus.Failed
   1151                 interactive_query.error_msg = str(e)
-> 1152             raise InteractiveEngineInternalError(str(e)) from e
   1153         else:
   1154             if self.eager():

InteractiveEngineInternalError: 'Create interactive instance failed.'

**Comment by acezen** [comment, marked as answer]
看日志好像是环境中没有 bash shell，请问你是在什么环境下跑的示例？

**Reply by default233** [reply]
您好，我是在 windows 下使用 wsl 在 ubuntu 20.04 的环境下跑的示例，其他脚本也能正常跑的，按日志看好像是换行符转义的问题，我在 gie 源码里没有找到对应的脚本，所以想请问下是哪里用到了哪个脚本，我看下是不是 fileformat 格式问题，或者您这边帮忙看下是不是有其他的问题
嗯，对了，示例中除了 交互这块，其他的内容都已经跑通过了

**Reply by acezen** [reply]
gie的启动脚本 https://github.com/alibaba/GraphScope/blob/main/interactive_engine/assembly/bin/giectl 你看看fix下fileformat，能否成功拉起。 如果成功的话，可以提交一个pr fix这个bug

**Reply by acezen** [reply]
gieclt这个脚本在库里是正常的，可能是你在Windows环境下git clone导致格式的转换，通过 `git config --global core.autocrlf input` 可以fix

FYI: https://stackoverflow.com/questions/29045140/env-bash-r-no-such-file-or-directory/29045187

**Reply by default233** [reply]
问题已解决，十分感谢🙏
